src/main/java/org/openhab/tools/analysis/checkstyle/api/AbstractStaticCheck.java

```diff
@@ -227,4 +220,21 @@ protected void logMessage(String filePath, int line, String fileName, String mes
         fireErrors(filePath);
         dispatcher.fireFileFinished(filePath);
     }
+
+    /**
+     * Parsed the content of a markdown file.
+     *
+     * @param fileText - Represents the text contents of a file
+     * @param blockTypes - the enabled block types
+     * @return The markdown node
+     */
+    protected Node parseMarkdown(FileText fileText, Set<Class<? extends Block>> blockTypes) {
+        Parser parser = Parser.builder().enabledBlockTypes(blockTypes).build();
+        Node readmeMarkdownNode = parser.parse(fileText.getFullText().toString());
+        return readmeMarkdownNode;
+    }
+
+    private InputStream getInputStream(FileText fileText) {
+        return new ByteArrayInputStream(fileText.getFullText().toString().getBytes(fileText.getCharset()));
```

Comment:
Maybe https://commons.apache.org/proper/commons-io/javadocs/api-2.4/org/apache/commons/io/input/CharSequenceInputStream.html I don't like the current code, but could not quickly find an easier way. It looks correct.